### PR TITLE
Surveys

### PR DESCRIPTION
**Pull request by chantal66**
adds survey functionality to the application. 
- Creates a  new Survey instance.
- adds survey routes.
- adds email template. 
- Creates and sends emails. 
- the survey is being saved in DB. 
- after clicking yes or no user is being redirected to thank page. (for now is just a message)
- add button for new surveys

### CODE REVIEW

File: client/public/index.html [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
-  <head>
-    <meta charset="utf-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
-    <meta name="theme-color" content="#000000">
-    <!--
+
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+  <meta name="theme-color" content="#000000">
```
Comment (kat3kasper) [inline client/public/index.html:7]: cool didn't even know you could do this

File: routes/surveyRoutes.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const surveyTemplate = require('../services/emailTemplates/surveyTemplate');
+
+const Survey = mongoose.model('surveys');
+
+module.exports = app => {
+  app.get('/api/surveys/thanks', (req, res) => {
+    res.send('Thanks for voting!');
+  });
+
+  app.post('/api/surveys', requireLogin, requireCredits, async (req, res) => {
```
Comment (kat3kasper) [inline routes/surveyRoutes.js:14]: This method is pretty long maybe you could have a SurveyHelper or something.

File: services/Mailer.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    this.body = new helper.Content('text/html', content);
+    this.recipients = this.formatAddresses(recipients); //specifies that im sending emails to multiple people
+
+    this.addContent(this.body);
+    this.addClickTracking(); // enabling click tracking
+    this.addRecipients();
+  }
+
+  formatAddresses(recipients) {
+    return recipients.map(({ email }) => {
```
Comment (kat3kasper) [inline services/Mailer.js:22]: 👍

File: routes/billingRoutes.js [hunk 4]
```diff
@@ -5,14 +5,14 @@ const requireLogin = require('../middlewares/requireLogin');
 module.exports = app => {
   app.post('/api/stripe', requireLogin, async (req, res) => {
     const charge = await stripe.charges.create({
-      amount: 500, // 500 cents = 5 dollars
+      amount: 500,
```
Comment (kat3kasper) [inline routes/billingRoutes.js:8]: I think the comment is helpful! Are users only able to buy in increments of 5 at a time?

**Review by kat3kasper — COMMENTED** [review]
Looking good. Clean readable code

File: client/public/index.html (hunk 1 above)
Comment (chantal66) [reply client/public/index.html:7]: :)

File: routes/billingRoutes.js (hunk 4 above)
Comment (chantal66) [reply routes/billingRoutes.js:8]: yes... I think after Turing.. I want users to be able to be refunded or put their own amount.

File: routes/surveyRoutes.js (hunk 2 above)
Comment (chantal66) [reply routes/surveyRoutes.js:14]: thank you for the feedback... I'll put a TODO for this. 😄

**Pull request merged**